app/src/main/java/fi/bitrite/android/ws/ui/FilterListFragment.java

```diff
@@ -0,0 +1,152 @@
+import fi.bitrite.android.ws.ui.util.HostFilterManager;
+
+public class FilterListFragment extends BaseFragment implements SeekBar.OnSeekBarChangeListener {
+
+    @Inject HostFilterManager mHostFilterManager;
+    @BindView(R.id.last_access_seekbar) SeekBar mLastAccessSeekBar;
+    @BindView(R.id.last_access_text) TextView mLastAccessText;
+    @BindView(R.id.currently_available_checkbox) CheckBox mCurrentlyAvailableCheckBox;
+
+    public static Fragment create() {
+        Bundle bundle = new Bundle();
+
+        Fragment fragment = new FilterListFragment();
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_filter_list, container, false);
+        ButterKnife.bind(this, view);
+        mHostFilterManager.activateFilter(HostFilterManager.RECENT_ACTIVITY_FILTER_KEY);
+        mLastAccessSeekBar.setOnSeekBarChangeListener(this);
+        mLastAccessSeekBar.setProgress(convertLastAccessDaysToProgress(
+              mHostFilterManager.getFilterValue(HostFilterManager.RECENT_ACTIVITY_FILTER_KEY)));
+        mLastAccessSeekBar.setMax(10);
+
+        mCurrentlyAvailableCheckBox.setChecked(
+            mHostFilterManager.isFilterActive(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY));
+        return view;
+    }
+
+    @OnClick(R.id.currently_available_checkbox)
+    void onCurrentlyAvailableClicked(View view) {
+        boolean checked = ((CheckBox) view).isChecked();
+        if (checked) {
+            mHostFilterManager.activateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+        } else {
+            mHostFilterManager.deactivateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+        }
+    }
+
+    @OnClick(R.id.btn_clear_filters)
+    void onClearFiltersClicked(View view) {
+        mLastAccessSeekBar.setProgress(10);
+        mCurrentlyAvailableCheckBox.setChecked(false);
+        mHostFilterManager.deactivateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+    }
+
+    @Override
+    protected CharSequence getTitle() {
+        return getString(R.string.title_fragment_filter_list);
+    }
+
+    @Override
+    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+      mLastAccessText.setText(convertLastAccessProgressToText(progress));
+      mHostFilterManager.updateFilterValue(HostFilterManager.RECENT_ACTIVITY_FILTER_KEY,
+          convertLastAccessProgressToDays(progress));
+    }
+    @Override
+    public void onStartTrackingTouch(SeekBar seekBar) { }
+    @Override
+    public void onStopTrackingTouch(SeekBar seekBar) { }
+
+    // [0, 1, ... 10] -> {1, 2, 3, 7, 14, 30, 60, 90, 180, 365, -1}.
+    private int convertLastAccessProgressToDays(int progress) {
+        switch (progress) {
```

Comment:
Doing mostly stuff in Python, I'm a huge fan of doing things explicitly. Just a reminder that this could be written in two lines without losing much readability.
```
List<Integer> days = Arrays.asList(1, 2, 3, 7, ..., -1);
return days.get(progress);
```

Reply:
Thanks. I find the original more readable, and marginally easier to update as the indices are explicitly aligned. That said, my preference is pretty arbitrary, and I'm happy to change it if you feel strongly about it :)

Reply:
A switch-case for a range from 0 to n feels very strange to me a. But I don't feel strong about it.